client/src/components/categorical/category/index.js

```diff
@@ -30,36 +31,49 @@ class Category extends React.Component {
     super(props);
     this.state = {
       isChecked: true,
+      categorySummary: this.createCategorySummary(),
     };
   }
 
   componentDidUpdate(prevProps) {
-    const { categoricalSelection, metadataField } = this.props;
+    const { categoricalSelection, metadataField, world } = this.props;
+    let { categorySummary } = this.state;
+
+    if (
+      world !== prevProps.world ||
+      metadataField !== prevProps.metadataField ||
+      !categorySummary
+    ) {
+      const newCategorySummary = this.createCategorySummary();
+      if (categorySummary !== newCategorySummary) {
+        categorySummary = newCategorySummary;
+        this.setState({ categorySummary }); // eslint-disable-line react/no-did-update-set-state
```

Comment:
I could be missing something, but what's your thought on listing `categorySummary` creation into `mapStateToProps()` inside `@connect`? And if `_createCategorySummary()` call is expensive, we could memoize it based on `world` and `metadataField`?

E.g.,

```ts
@connect((state, ownProps) => {
  const { metadataField } = ownProps;
  const { world } = state

  return {
    world,
    isColorAccessor: state.colors.colorAccessor === metadataField,
    categoricalSelection: state.categoricalSelection,
    categorySummary: categorySummary(world, metadataField), // <-- NEW
    annotations: state.annotations,
    universe: state.universe,
    schema: state.world?.schema,
  };
})
```

Reply:
sure, that would work today, but it won't work with the (soon to land) asynchronous data loading that motivated this PR.  So I would have to revert that change almost immediately :-)

TL;DR - I'd rather leave it in state, which is where it must be soon.

Reply:
hmm since Redux's `connect()` is triggered whenever the Redux store has an update, creating `categorySummary` could still happen in `mapStateToProps()` instead of in the view component?

What I'm thinking is, if Redux state doesn't have the needed data, `categorySummary(world, metadataField)` will return something boring, like `null` or `undefined`. But once the store is updated, `categorySummary(world, metadataField)` will automatically get called again via `connect()`. So it should work in async model too?

Reply:
true, but it would be wasted work, as it only needs to be run when the data changes.

Currently that "data" is in the reducer state, but very soon it won't be.